database/migrations/2018_01_10_080009_create_qrcodes_table.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class CreateQrcodesTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('qrcodes', function (Blueprint $table) {
```

Comment:
- what's qrcodes table purpose?

Reply:
save all of QRcode of book and status is checked QRcode printed or not.

Reply:
- I understood for this table. but I have some suggetion:
   + you should delete QRcode column in book table
   + you shoud reference book_id column
   + you should save only code_id in database, prefix was defined in model.